src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * Ably Batch client based on REST API
+ */
+public class DefaultAblyBatchClient implements AblyClient {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultAblyBatchClient.class);
+
+    protected final ChannelSinkMapping channelSinkMapping;
+    protected final MessageSinkMapping messageSinkMapping;
+    protected final ChannelSinkConnectorConfig connectorConfig;
+
+    private final ConfigValueEvaluator configValueEvaluator;
+    AblyRest restClient;
+
+    public DefaultAblyBatchClient(ChannelSinkConnectorConfig connectorConfig, ChannelSinkMapping channelSinkMapping,
+                                  MessageSinkMapping messageSinkMapping, ConfigValueEvaluator configValueEvaluator) throws AblyException {
+        this.connectorConfig = connectorConfig;
+        this.channelSinkMapping = channelSinkMapping;
+        this.messageSinkMapping = messageSinkMapping;
+        this.configValueEvaluator = configValueEvaluator;
+        this.restClient = new AblyRest(connectorConfig.getPassword(CLIENT_KEY).value());
+    }
+
+    /**
+     * Function that uses the REST API client to send batches.
+     * @param records list of sink records.
+     * @throws ConnectException
+     */
+    @Override
+    public void publishBatch(List<SinkRecord> records) throws ConnectException {
+
+        if(!records.isEmpty()) {
+            List<BatchSpec> batchSpecs = new ArrayList<>();
+            Map<String, List<Message>> groupedMessages = this.groupMessagesByChannel(records);
+            groupedMessages.forEach((key, value) -> {
+                batchSpecs.add(new BatchSpec(Set.of(key), value));
+            });
+            try {
+                logger.info("Ably BATCH call -Thread(" + Thread.currentThread().getName() + ")");
+                this.sendBatches(batchSpecs);
+            } catch (Exception e) {
+                logger.error("Error while sending batch", e);
+            }
+        }
+    }
+
+    /**
+     * Function to group the Ably Message objects by channel Name.
+     * @param sinkRecords
+     * @return
+     */
+    public Map<String, List<Message>> groupMessagesByChannel(List<SinkRecord> sinkRecords) {
+
+        HashMap<String, List<Message>> channelNameToMessagesMap = new HashMap();
+
+        for(SinkRecord record: sinkRecords) {
+            try {
+                if (!shouldSkip(record)) {
+                    final String channelName = channelSinkMapping.getChannelName(record);
+                    final Message message = messageSinkMapping.getMessage(record);
+
+                    final List<Message> updatedMessages = channelNameToMessagesMap.getOrDefault(channelName, new ArrayList<>());
```

Comment:
@jaley I made the change here, for some reason, github doesnt update it in the original thread.